### DISCUSSION TITLE
Backups error

### DISCUSSION THREAD

**Discussion by gmlanz** [vmware-tanzu/velero#6437] [Community support Q&A]
I've recently installed Velero but I'm getting issues accessing the blob storage.
The error messages are:
level=error msg="Error getting a backup store" backup-storage-location=velero/backups-primary controller=backup-storage-location error="rpc error: code = Unknown desc = config has invalid keys [region]; valid keys are [resourceGroup storageAccount subscriptionId blockSizeInBytes storageAccountKeyEnvVar credentialsFile bucket prefix caCert]" error.file="/go/pkg/mod/github.com/vmware-tanzu/velero@v1.7.1/pkg/plugin/framework/validation.go:50" error.function=github.com/vmware-tanzu/velero/pkg/plugin/framework.validateConfigKeys logSource="pkg/controller/backup_storage_location_controller.go:148"
time="2023-06-29T12:35:07Z" level=info msg="BackupStorageLocation is invalid, marking as unavailable" backup-storage-location=velero/backups-primary controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:131"
time="2023-06-29T12:35:07Z" level=error msg="Current BackupStorageLocations available/unavailable/unknown: 0/2/0, BackupStorageLocation \"backups-primary\" is unavailable: rpc error: code = Unknown desc = config has invalid keys [region]; valid keys are [resourceGroup storageAccount subscriptionId blockSizeInBytes storageAccountKeyEnvVar credentialsFile bucket prefix caCert])" controller=backup-storage-location logSource="pkg/controller/backup_storage_location_controller.go:192"

My credentials-velero file looks like this:
AZURE_BACKUP_RESOURCE_GROUP="velero-backups-001"                                                                                                                               
AZURE_BACKUP_SUBSCRIPTION_ID="xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"                                                                                                                                                               
AZURE_BACKUP_SUBSCRIPTION_NAME="UK-Dev1"                                                                                                                                                             
AZURE_CLIENT_ID="xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"                                                                                                                        
AZURE_CLIENT_SECRET="xxxxxxxxxxxxxxxxxxxxxxxxxx"                                                                                                                    
AZURE_CLOUD_NAME="AzurePublicCloud"                                                                                                                                                  
AZURE_RESOURCE_GROUP="rg-dev-xxx-xxx-001"                                                                                                                                          
AZURE_STORAGE_ACCOUNT_ACCESS_KEY="xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"                                                                    
AZURE_STORAGE_ACCOUNT_ID="devvelero001"                                                                                                                                           
AZURE_SUBSCRIPTION_ID="xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"                                                                                                                        
AZURE_SUBSCRIPTION_NAME="UK-Dev1"                                                                                                                              
AZURE_TENANT_ID="xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"                                                                                                                        
BLOB_CONTAINER="velero"

The install command I used was this:
velero install --provider azure --plugins velero/velero-plugin-for-microsoft-azure:v1.6.0 --bucket $BLOB_CONTAINER --secret-file ./credentials-velero.txt --backup-location-config resourceGroup=$AZURE_BACKUP_RESOURCE_GROUP,storageAccount=$AZURE_STORAGE_ACCOUNT_ID,storageAccountKeyEnvVar=AZURE_STORAGE_ACCOUNT_ACCESS_KEY[,subscriptionId=$AZURE_BACKUP_SUBSCRIPTION_ID]

This is just for a test. I'm sure it is something simple that I'm missing, any help from Velero experts greatly appreciated.

**Comment by 0x113** [comment]
It looks to me that the credentials file has too many values. If you want to use Azure Service Principal and use the same subscription for backups, the following format should work:

Credentials file:
```
AZURE_SUBSCRIPTION_ID=<subscription-id>
AZURE_TENANT_ID=<tenant-id>
AZURE_CLIENT_ID=<client-id>
AZURE_CLIENT_SECRET=<secret>
AZURE_RESOURCE_GROUP=<cluster-infrastructure-resource-group>
AZURE_CLOUD_NAME=AzurePublicCloud
```

Velero command:
```
velero install \    
    --provider azure \
    --plugins velero/velero-plugin-for-microsoft-azure:v1.6.0 \                                    
    --bucket $BLOB_CONTAINER \
    --secret-file ./credentials-velero \
    --backup-location-config resourceGroup=$AZURE_BACKUP_RESOURCE_GROUP,storageAccount=$AZURE_STORAGE_ACCOUNT_ID,subscriptionId=$AZURE_BACKUP_SUBSCRIPTION_ID
```

If you want to take file system backups, you should also use `use-node-agent` option and specify ``uploader type``. For the second parameter you can either use `restic` or `kopia`.
